packages/react-sdk/src/helpers/messagesDb.ts

```diff
@@ -0,0 +1,39 @@
+import type { DecodedMessage } from "@xmtp/xmtp-js";
+import type { Table } from "dexie";
+import Dexie from "dexie";
+import { getConversationId } from "./getConversationId";
+
+export type CachedMessage = {
+  id: string;
+  cId: string;
+  bytes: Uint8Array;
+  sent: Date;
+};
+
+export class MessagesDB extends Dexie {
+  messages!: Table<CachedMessage>;
+
+  constructor() {
+    super("__XMTP__");
+    this.version(1).stores({
+      messages: "id, cId, sent",
+    });
+  }
+
+  // persist encrypted message to cache
+  async persistMessage(message: DecodedMessage) {
+    const { id, sent } = message;
+    await this.messages.put(
+      {
+        id,
+        cId: getConversationId(message.conversation),
+        // encrypted message
```

Comment:
To clarify, this isn't actually encrypted. `DecodedMessage.toBytes()` is a proto representation of the decrypted message. 

If we wanted to encrypt these messages at rest, we should look into ways of having a static encryption key (maybe the identity key)

Reply:
oops, thanks for the clarification! hmmm, are there concerns that messages are not encrypted at rest?

Reply:
We should think about encryption as a fast-follow. XMTP already has quite a few keys that could be repurposed as an encryption key for the DB, so that should be doable.

https://github.com/mark43/dexie-encrypted has potential. Never tried it.